task/dump-agent/src/main.rs

```diff
@@ -20,83 +20,48 @@ use userlib::*;
 const_assert!(DUMP_READ_SIZE & (DUMP_READ_SIZE - 1) == 0);
 const_assert!(DUMP_READ_SIZE <= 1024);
 
-struct ServerImpl {
-    areas: [DumpArea; 3],
-}
+struct ServerImpl;
 
 #[cfg(not(feature = "no-rot"))]
 task_slot!(SPROT, sprot);
 
+task_slot!(JEFE, jefe);
+
 impl ServerImpl {
-    fn initialize(&self) {
-        let mut next = 0;
-
-        for area in self.areas.iter().rev() {
-            unsafe {
-                let header = area.address as *mut DumpAreaHeader;
-
-                //
-                // We initialize our dump header with deliberately bad magic
-                // to prevent any dumps until we have everything initialized
-                //
-                (*header) = DumpAreaHeader {
-                    magic: DUMP_UNINITIALIZED,
-                    address: area.address,
-                    nsegments: 0,
-                    written: size_of::<DumpAreaHeader>() as u32,
-                    length: area.length,
-                    agent_version: DUMP_AGENT_VERSION,
-                    dumper_version: DUMPER_NONE,
-                    next,
-                }
-            }
+    fn initialize(&self) -> Result<(), DumpAgentError> {
+        let jefe = Jefe::from(JEFE.get_task_id());
+        jefe.initialize_dump_areas()
```

Comment:
Right now, this means that `jefe` can't dump crashing tasks until `dump-agent` starts up.

How about letting `jefe` initialize the dump areas itself, then call `Jefe.get_state` here (which will only return once `jefe` enters the Idol runtime loop)?

Reply:
We also use this entry point to reinitialize the dump areas -- but you're absolutely right that Jefe should do this on its own without relying on the dump agent.  Will change the name of the Jefe entry point to make clear that a caller can reinitialize the dump areas.

Reply:
Done